src/components/Cluster/NewCluster/ReleaseSelector.js

```diff
@@ -23,22 +23,37 @@ const FlexRowDiv = styled.div`
   }
 `;
 
+const MarginlessReleaseComponentLabel = styled(ReleaseComponentLabel)`
+  margin: 0;
+`;
+
 class ReleaseSelector extends React.Component {
   state = {
     kubernetesVersion: '',
     loading: true,
   };
 
   componentDidMount() {
-    const kubernetesVersion = this.props.selectedRelease
-      ? this.props.releases[this.props.selectedRelease].components.find(
-          (component) => component.name === 'kubernetes'
-        ).version
-      : undefined;
+    this.setState({
+      kubernetesVersion: this.extractKubernetesVersion(),
+      loading: false,
+    });
+  }
 
-    this.setState({ kubernetesVersion, loading: false });
+  componentDidUpdate({ selectedRelease: prevRelease }) {
+    if (prevRelease !== this.props.selectedRelease) {
+      this.setState({ kubernetesVersion: this.extractKubernetesVersion() });
+    }
```

Comment:
Not a blocker from me but this is the perfect use case for React's built-in `gerDerivedStateFromProps`
https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

Also, the `loading` state can be `false` by default

Reply:
Now that I think about it more, I think we can actually remove the `kubernetesVersion` property from the local state completely.
The release version could be extracted in the `render()` method, and the `extractKubernetesVersion` method could be memoized, to prevent unnecessary updates.

What do you think?